src/utils/editor/source-documents.js

```diff
@@ -126,7 +126,12 @@ function showSourceText(
   if (hasDocument(source.id)) {
     const doc = getDocument(source.id);
     if (editor.codeMirror.doc === doc) {
-      editor.setMode(getMode(source, symbols));
+      const mode = getMode(source, symbols);
+
+      if (doc.mode.name !== mode.name) {
```

Comment:
Nice idea to check probably saves a good amount of work too!

Reply:
yeah, a ton of time